Source/DynamicFeatureRegistry.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the MIT license. See LICENSE file in https://github.com/yahoo/Override for terms.
+
+import Foundation
+
+/// DynamicFeatureRegistry a subclass of FeatureRegistry which provides an interface for using
+/// a collection of dynamic features instead of just pre-defined ones. When a feature is added,
+/// there is an attempt to apply an override state from the underlying data FeatureStore.
+@objc open class DynamicFeatureRegistry: FeatureRegistry {
+
+    /// A local store of the current dynamic features
+    private var dynamicFeatures: [String : AnyFeature] = [:]
+
+    override var features: [LabeledItem] {
+        let dynamicLabeledItems = dynamicFeatures.values.reduce(into: []) { (result, feature) in
+            result.append(LabeledFeatureItem(label: feature.key, feature: feature))
+        }
+        let staticFeatures = super.features
+        /// merge both dynamic and static features
+        let allFeatures = dynamicLabeledItems + staticFeatures
```

Comment:
I did read the docs at the top, but it seems arbitrary to merge static features _into_ dynamic features rather than the other way around. Should this use-case specific assumption be delegated to the user?

Perhaps something akin to Swift Dictionary's merge/merging into which provides a `uniquing:` block

Reply:
I'm combining the two data sets so both can be displayed and managed in the UI - Is the suggestion here to do `staticFeatures + dynamicLabeledItems` instead?

Reply:
Not sure if there is one right way to do it, so Adam's suggestion of a closure to handle the merge is interesting. (ala https://developer.apple.com/documentation/swift/dictionary/3127171-merge)

An example of where I could see this being useful is to report name conflicts on that merge with instrumentation events.